src/hotspot/share/jfr/periodic/sampling/jfrThreadSampler.cpp

```diff
@@ -387,6 +388,7 @@ bool JfrThreadSampleClosure::do_sample_thread(JavaThread* thread, JfrStackFrame*
   }
 
   bool ret = false;
+  StackWatermarkSet::start_processing(thread, StackWatermarkKind::gc);
```

Comment:
Is this now needed unconditionally, for all threads sampled? What is the overhead of introducing it?

Reply:
Compared to everything else that thread sampling does, I really wouldn't worry about it. In the common case it will just check if processing already started, and then continue.

Reply:
The reason for asking is for upcoming changes that try to minimize everything involved in sampling. Is this only needed for virtual threads?

Reply:
I see. It is indeed only needed for virtual threads. If you are going to perform very frequent sampling, then it's worth mentioning that the check if the processing already started requires load_acquire on non-TSO platforms. I suppose that would be the dominating cost. But for this load_acquire to start showing up as a significant overhead, I think you would have to do sampling in the order of magnitude of hundreds of nanosecond interval levels. So unless you are getting to such intense frequencies, I think I would still not worry about it. What do you think?

Reply:
Thanks for the information, Erik. I think we can go with your solution, I will take the information about Virtual Threads into account for later, there is still a lot of magic left that needs to be done in the new sampling system in order to incorporate them.